### PR TITLE
Initial WebSocket support

### PR DESCRIPTION
**Pull request by gatzjames**
Decision to make before merge:
- [x] timeline format -> array -> newline #5124
- [x] review/simplify preload api shape/ipc channel naming #5125
- [x] decision about consistent model and export naming conventions 
- [x] persisted payloads
- [x] websocketrequest data model review
- [x] websocketresponse/response data model review -> bodyPath -> eventLogPath
- [x] event log bug

Nitpicks
- [x] reverse event log
- [x] dupe connection success race condition
- [x] remove send message event from timeline
- [x] rename icn-ellipsis-circle2.svg -> icn-checkmark-circle.svg

naming decision notes:
- ~~two words "WebSocket", "web-socket", "web_socket", "webSocket"~~
- ~~one word, with an abbreviation exception "WebSocket", "websocket", "WS"~~
- redux implementation forces us to be inconsistent, it lowercases the first letter of the model type and forces us to set the model type to WebsocketRequest creating an exception. Proposal -> #5158 
- default decision is for filenames to be treat websocket/s as one word and variables and types to treat W/webSocket as two words.

### CODE REVIEW

File: packages/insomnia/src/models/helpers/request-operations.ts [hunk 1]
```diff
@@ -1,31 +1,57 @@
...
   // @ts-expect-error -- TSCONVERSION
-    : models.request.update(request, patch);
+  if (isWebSocketRequest(request)) {
+    // @ts-expect-error -- TSCONVERSION
+    return models.websocketRequest.update(request, patch);
+  } else {
+    // @ts-expect-error -- TSCONVERSION
+    return models.request.update(request, patch);
+  }
 }
```
Comment (jackkav) [inline packages/insomnia/src/models/helpers/request-operations.ts:41]: unused else clause

File: packages/insomnia/src/ui/components/dropdowns/websocket-request-actions-dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        <i className="fa fa-trash-o" /> Delete
+        <DropdownHint keyBindings={hotKeyRegistry[hotKeyRefs.REQUEST_SHOW_DELETE.id]} />
+      </DropdownItem>
+
+      <DropdownDivider />
+    </Dropdown>
+  );
+});
+
+WebSocketRequestActionsDropdown.displayName = 'RequestActionsDropdown';
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/dropdowns/websocket-request-actions-dropdown.tsx]:
```suggestion
WebSocketRequestActionsDropdown.displayName = 'WebSocketRequestActionsDropdown';
```

File: packages/insomnia/src/ui/components/sidebar/sidebar-children.tsx [hunk 3]
```diff
@@ -50,16 +51,37 @@ export const SidebarChildren: FC<Props> = ({
...
 
     return (
       <>
-        {rows.map(row => (!isInPinnedList && row.hidden)
-          ? null
-          : (isRequest(row.doc) || isGrpcRequest(row.doc))
-            ? (
+        {rows
+          .filter(row => !(!isInPinnedList && row.hidden))
+          .map(row => {
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/sidebar/sidebar-children.tsx:65]: 💯

**Review by filfreire — APPROVED** [review]
🎉 
https://www.youtube.com/watch?v=T-v20t_puwE

**Review by jackkav — APPROVED** [review]

File: packages/insomnia/src/ui/components/websockets/websocket-request-pane.tsx [hunk 4]
```diff
@@ -52,16 +52,30 @@ const PaneHeader = styled(OriginalPaneHeader)({
...
   request,
-  payloadType,
+  previewMode,
+  initialValue,
+  createOrUpdatePayload,
   environmentId,
 }) => {
   const editorRef = useRef<UnconnectedCodeEditor>(null);
+  useEffect(() => {
+    let isMounted = true;
```
Comment (marckong) [inline packages/insomnia/src/ui/components/websockets/websocket-request-pane.tsx:71]: I wonder what was the reason for this check for isMounted?

**Pull request merged**